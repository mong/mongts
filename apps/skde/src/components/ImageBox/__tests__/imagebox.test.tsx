/**
 * To view a rendered version of all the snapshots:
 *
 * Copy the contents of ./__snapshots__/index.tsx.snap
 * Go to https://codesandbox.io/s/snapshot-viewer-lqjjsm?file=/src/snapshots.js
 * Replace the contents with what you copied
 */

import { render } from "@testing-library/react";
import { ImageBox } from "..";

test("Standard render", async () => {
  const { container } = render(
    <ImageBox
      caption=""
      alt=""
      source="https://picsum.photos/id/237/100/100.jpg"
      width={400}
      height={250}
      title=""
    />
  );
  expect(container).toMatchSnapshot();
});

test("Render with some more text", async () => {
  const { container } = render(
    <ImageBox
      caption="Dette er en test"
      alt="Tilfeldig bilde fra picsum"
      source="https://picsum.photos/id/237/300/200"
      width={300}
      height={200}
      title="Æ Å"
    />
  );
  expect(container).toMatchSnapshot();
});
