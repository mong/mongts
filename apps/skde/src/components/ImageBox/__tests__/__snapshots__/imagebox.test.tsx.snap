// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Render with some more text 1`] = `
<div>
  <figure
    style="display: flex; flex-direction: column;"
  >
    <img
      alt="Tilfeldig bilde fra picsum"
      data-nimg="1"
      decoding="async"
      height="200"
      loading="lazy"
      src="https://picsum.photos/id/237/300/200?w=640&q=75"
      srcset="https://picsum.photos/id/237/300/200?w=384&q=75 1x, https://picsum.photos/id/237/300/200?w=640&q=75 2x"
      style="color: transparent;"
      title="Æ Å"
      width="300"
    />
    <figcaption>
      Dette er en test
    </figcaption>
  </figure>
</div>
`;

exports[`Standard render 1`] = `
<div>
  <figure
    style="display: flex; flex-direction: column;"
  >
    <img
      alt=""
      data-nimg="1"
      decoding="async"
      height="250"
      loading="lazy"
      src="https://picsum.photos/id/237/100/100.jpg?w=828&q=75"
      srcset="https://picsum.photos/id/237/100/100.jpg?w=640&q=75 1x, https://picsum.photos/id/237/100/100.jpg?w=828&q=75 2x"
      style="color: transparent;"
      title=""
      width="400"
    />
  </figure>
</div>
`;
